### PR TITLE
[BACKLOG-36850] Focus Visible - Accessibility -  PUC Home Page – Manage Data Sources

### PR DESCRIPTION
**Pull request by varuntangirala**
@pentaho/wcag , Please review

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 46s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl gwt/impl
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

File: gwt/impl/src/main/java/org/pentaho/ui/xul/gwt/tags/GwtButton.java [hunk 1]
```diff
@@ -288,6 +294,9 @@ public boolean onKeyDownPreview( char key, int modifiers ) {
             };
             menuBar = new MenuBar( true );
             menuBar.setAutoOpen( true );
+            menuBar.getElement().setId( DOM.createUniqueId() );
+            imageButton.getElement().setAttribute("aria-controls", menuBar.getElement().getId() );
+
```
Comment (dcleao) [inline gwt/impl/src/main/java/org/pentaho/ui/xul/gwt/tags/GwtButton.java:299]: Please, check formatting, and add:
```java
imageButton.getElement().setAttribute( "aria-expanded", "true" );

popupPanel.addCloseHandler( ev -> {
  imageButton.getElement().removeAttribute( "aria-controls" );
  imageButton.getElement().setAttribute( "aria-expanded", "false" );
} );
```

File: gwt/impl/src/main/java/org/pentaho/ui/xul/gwt/tags/GwtButton.java [hunk 2]
```diff
@@ -558,11 +567,24 @@ public void execute() {
...
         GwtButton.this.getXulDomContainer().invoke( command, new Object[] {} );
       } catch ( XulException e ) {
         System.out.println( "Error invoking method " + command + " " + e.getMessage() );
         e.printStackTrace();
       }
     }
+
+    private native void focusSync( com.google.gwt.dom.client.Element elem )/*-{
+      elem.focus();
+    }-*/;
```
Comment (dcleao) [inline gwt/impl/src/main/java/org/pentaho/ui/xul/gwt/tags/GwtButton.java:588]: Please, move this to `ElementUtils` (on the GWT project) as a public static method.

**Review by dcleao — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
### :x: Build failed in 42s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl gwt/impl
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — APPROVED** [review]

**Pull request merged**